### PR TITLE
assignment 3

### PR DESCRIPTION
**Pull request by ankitsrivastava777**

### CODE REVIEW

File: models/AccessToken.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "user",
+        unique: true,
+      },
+      access_token: {
+        type: String,
+      },
+      expiry: {
+        type: Date,
+        expires: "60s",
```
Comment (iamprashanttyagi) [inline models/AccessToken.js]: expire 60 sec ?

File: models/UserAddress.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      },
+      phone_no: {
+        type: String,
+      },
+      userdata: [
+        {
+          type: mongoose.Schema.Types.ObjectId,
+          ref: "user_details",
+        },
+      ],
```
Comment (iamprashanttyagi) [inline models/UserAddress.js]: use of `userdata` ?

File: routes/user.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  user.findOne({ username: req.body.username }, async function (err, results) {
+    var pass = results.password.toString();
+    var userId = results._id;
+    var input_password = pass.toString();
+    var user_password = req.body.password;
+    var tokenId = userId.toString();
+    if (await bcrypt.compare(user_password, input_password)) {
+      const pwd = passwordHash.generate(req.body.password);
+      AccessToken.findOne({ user_id: userId }, function (err, userDetails) {
+        if (userDetails && userDetails._id) {
```
Comment (iamprashanttyagi) [inline routes/user.js]: details from the access token table will not be userDetails so provide relevant name to the variable

File: routes/user.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+          res.status(500).json({
+            error: 1,
+            message: "already login",
+            data: null,
+          });
```
Comment (iamprashanttyagi) [inline routes/user.js]: if details exits of token then provide that token

File: routes/user.js [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+              res.status(200).json({
+                error: 1,
+                message: "token not saved",
+                data: null,
+              });
+            }
+            res.status(200).json({
+              error: 0,
+              message: "token saved",
+              data: tokenId,
```
Comment (iamprashanttyagi) [inline routes/user.js:78]: send token not any Id

File: routes/user.js [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+      id: req.user._id,
+      username: req.user.username,
+      email: req.user.email,
+    },
+  });
+});
+
+app.put("/delete/", auth, async function (req, res) {
+  var user_id = req.user.user_id;
+  await user.deleteOne({ _id: user_id }, function (err, results) {
```
Comment (iamprashanttyagi) [inline routes/user.js]: - don't use await and callback at same time.
- delete user address after user delete

File: routes/user.js [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+  user
+    .find()
+    .skip(skip_user_list)
+    .limit(pages_number)
+    .exec(function (err, userData) {
+      if (err) {
+        res.status(500).json({
+          message: "no data found",
+        });
+      }
```
Comment (iamprashanttyagi) [inline routes/user.js]: send same error response format

**Review by iamprashanttyagi — CHANGES_REQUESTED** [review]

File: models/AccessToken.js (hunk 1 above)
Comment (ankitsrivastava777) [reply models/AccessToken.js]: added for token expire in 60s will change it to 10min

File: models/UserAddress.js (hunk 2 above)
Comment (ankitsrivastava777) [reply models/UserAddress.js]: removed

File: routes/user.js [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+  res.status(200).json({
+    error: 0,
+    message: "user deleted",
+    data: null,
+  });
+});
+
+app.get("/list/:limit/:page", function (req, res) {
+  pages_number = Number(req.params.page);
+  limit = req.params.limit;
```
Comment (iamprashanttyagi) [inline routes/user.js]: use `let` or `var` or `const` to declare the variable

**Review by iamprashanttyagi — CHANGES_REQUESTED** [review]

**Pull request merged**